src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDataReader.cs

```diff
@@ -1994,7 +1994,7 @@ override public TextReader GetTextReader(int i)
                 {
                     encoding = _metaData[i].encoding;
                 }
-
+                encoding = mt.SqlDbType == SqlDbTypeExtensions.Json ? System.Text.Encoding.UTF8 : encoding;
```

Comment:
✏️ nit, but I usually find it easier to read if this kind of behavior is expressed as mutually exclusive cases. This way it's clear that there's not something special about the relationship between NCharType and Json that would further change the outcome. Even better would be a switch statement, but then we'd have to unravel the IsNCharType determination.

Something like:
```c#
System.Text.Encoding encoding;
if (mt.SqlDbType == SqlDbTypeExtensions.Json) {
    encoding = System.Text.Encoding.UTF8
}
else if (mt.IsNCharType)
{
    // NChar types always use unicode
    encoding = SqlUnicodeEncoding.SqlUnicodeEncodingInstance;
}
else
{
    encoding = _metaData[i].encoding;
}
```

Reply:
Updated with suggested changes.